### DISCUSSION TITLE
Authorization headers overwritten by Safari when using .htaccess. Custom header possible?

### DISCUSSION THREAD

**Discussion by ivo-w3** [adonisjs/core#4105] [Help] [answered]
When using a .htaccess, Safari will overwrite the Authorization header with the user-input from the htaccess promt, and thus AdonisJS 5 cant authorize the requests. Is there any way in the auth (api) config or some middleware to change AdonisJS 5 to use some custom header?

**Comment by thetutlage** [comment, marked as answer]
You can create a middleware and put it before the auth middleware. Inside that middleware you can read the value from your custom header and set it as `Authorization` header

**Reply by ivo-w3** [reply]
Hey thank you! I had the same intention but I did not find the original request I did it like this:

request.request.headers.authorization = request.header("x-custom-token");
